### PR TITLE
Sage/a guide to building decentralized loan dapp

### PR DESCRIPTION
**Pull request by dahnny**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.

> NOTE: We cannot review your Pull Request if it does not meet the standard outlined above.
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+I dont want to bore you with my love for the remix IDE but I do love remix for writing my smart contracts and that is what we would be using in this tutorial.
+
+Here is a preview of the Remix IDE:
+![image](images/2.png)
+
+On Remix, We would create a new workspace and then a new file which we would name `loan.sol`
+
+Starting in the first line, you include a statement that specifies the license under which the code is being released.
+
+```js
```
Comment (developerkunal) [inline blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md:52]: change all solidity code block to ```solidity

File: blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+        return false;
+    }
+
+    function getLoanLength() public view returns (uint256) {
+        return loanLength;
+    }
+}
+```
+
+I am going to do something different in this tutorial by leaving the implementation of the frontend as a challenge. You can use whatever framework of your choice, you can also check out my previous tutorial to get some ideas. [Check It Out]((https://docs.celo.org/blog/tutorials/how-to-build-car-marketplace-dapp-with-react)).
```
Comment (developerkunal) [inline blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md:500]: `Fix The Brackets here [Check It Out](https://docs.celo.org/blog/tutorials/how-to-build-car-marketplace-dapp-with-react). make it like this`

File: blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+The function first checks if the cUSD token is transferred from the sender to the bank's address. If this transfer fails, an error message "Error during Loan Registration" will be displayed.
+
+If the transfer is successful, the function will create a new `LoanDetails` struct instance with the borrower's details, status set to "review," and other loan-related information, such as the loan amount, tenure, and timestamp.
+
+After that, the new `LoanDetails` instance is stored in the `loan` mapping with a new unique identifier assigned to it, and the loan length counter is incremented by one. This indicates that a new loan has been successfully registered.
+
+
+The next function on the list is the `getLoans` function that allows the admin to view the loan details associated with a given index. The function takes in an index as input and checks if the caller is an admin.
+
+``solidity
```
Comment (developerkunal) [inline blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md:214]: here you forgot ```solidity

Comment (dahnny) [reply blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md]: Done

**Review by Ishanpathak1 — APPROVED** [review]
Approved @developerkunal

**Pull request merged**